sycl/plugins/unified_runtime/ur/adapters/cuda/device.cpp

```diff
@@ -0,0 +1,1183 @@
+//===--------- device.cpp - CUDA Adapter -----------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-----------------------------------------------------------------===//
+
+#include <cassert>
+
+#include "context.hpp"
+#include "device.hpp"
+#include "platform.hpp"
+
+int getAttribute(ur_device_handle_t device, CUdevice_attribute attribute) {
+  int value;
+  sycl::detail::ur::assertion(
```

Comment:
There are also cases like these, where `assertion` is used instead of `check_error`. This will also lead to lost native error information. All such cases should be setting the last message and reporting a plugin specific error as described here: https://github.com/oneapi-src/unified-runtime/issues/500#issuecomment-1554854496